rel-sign.sh

```diff
@@ -0,0 +1,137 @@
+    -r, --repo - repository, in the format username/repository
+    -v, --version - version, in the format x.y.z, without leading v
+    -k, --key - signing key id, fingerprint or email
+    -gpg - use gpg instead of rnp for signing.
+    -s, --src - use specified folder for release sources comparison instead of downloading from GitHub.
+    -pparams - addition OpenPGP params, like '--homedir .rnp', '--keyfile seckey.asc', etc.\n"
+}
+
+# Extract parameters
+if [ $# -eq 0 ]; then
+    print_help
+    exit 0
+fi
+
+PPARAMS=""
+
+while [ $# -gt 0 ]; do
+  case "$1" in
+    --repo*|-r*)
+      if [[ "$1" != *=* ]]; then shift; fi # Value is next arg if no `=`
+      REPO="${1#*=}"
+      REPOLAST=`basename ${REPO}`
+      ;;
+    --version*|-v*)
+      if [[ "$1" != *=* ]]; then shift; fi
+      VERSION="${1#*=}"
+      ;;
+    --key*|-k*)
+      if [[ "$1" != *=* ]]; then shift; fi
+      KEY="${1#*=}"
+      ;;
+    --gpg)
+      USEGPG=1
+      ;;
+    --src*|-s*)
+      if [[ "$1" != *=* ]]; then shift; fi
+      SRCDIR=`realpath "${1#*=}"`
+      if [ ! -d ${SRCDIR} ]; then
+          printf "Directory ${SRCDIR} doesn't exist.\n"
+          exit 1
+      fi
+      printf "Comparing with sources from ${SRCDIR}\n"
+      ;;
+    --pparams*)
+      if [[ "$1" != *=* ]]; then shift; fi
+      PPARAMS="${1#*=}"
+      ;;
+    --help|-h)
+      print_help
+      exit 0
+      ;;
+    *)
+      >&2 printf "Invalid argument: $1\nUse --help or -h to get list of available arguments.\n"
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+# Check whether all parameters are specified.
+if [ -z ${REPO+x} ]; then
+    printf "Please specify repository via -r or --repo argument.\n"
+    exit 1
+fi
+if [ -z ${VERSION+x} ]; then
+    printf "Please specify release version via -v or --version argument.\n"
+    exit 1
+fi
+if [ -z ${KEY+x} ]; then
+    printf "Signing key was not specified - so default one will be used.\n"
+fi
+
+# Fetch repository and release tarball/zip.
+printf "Fetching repository and releases...\n"
+set -x
+TMPDIR=$(_mktemp)
+pushd ${TMPDIR} > /dev/null
+
+if [ -z ${SRCDIR+X} ]; then
+    git clone https://github.com/${REPO}
+    pushd ${REPOLAST} > /dev/null
+    git checkout v${VERSION}
+    popd > /dev/null
+    SRCDIR=${REPOLAST}
+fi
+
+# Check .tar.gz
+wget https://github.com/${REPO}/archive/refs/tags/v${VERSION}.tar.gz
+tar xf v${VERSION}.tar.gz
+diff -qr --exclude=".git" ${SRCDIR} ${REPOLAST}-${VERSION}
+rm -rf ${REPOLAST}-${VERSION}
+
+# Check .zip
+wget https://github.com/${REPO}/archive/refs/tags/v${VERSION}.zip
+unzip -qq v${VERSION}.zip
+diff -qr --exclude=".git" ${SRCDIR} ${REPOLAST}-${VERSION}
+rm -rf ${REPOLAST}-${VERSION}
+popd > /dev/null
+
+# Sign
+KEYCMD=""
+if [ -z ${USEGPG+x} ]; then
+    # Using the rnp - default
+    if [ ! -z ${KEY+x} ]; then
+        KEYCMD="-u ${KEY}"
+    fi
+    rnp --sign --detach --armor ${KEYCMD} ${PPARAMS} ${TMPDIR}/v${VERSION}.tar.gz --output v${VERSION}.tar.gz.asc
```

Comment:
Since we are using Bash which supports arrays, it might be beneficial to use them in place of plain string for `$PPARAMS`, like so:

` rnp ... "${PPARAMS[@]}" ... `

which helps preserve any contiguous white spaces in the segments within `$PPARAMS`.

This means `PPARAMS=()` instead of `PPARAMS=""` for its initialization, and maybe require `--pparams` to be specified last in the command line so it is able to consume all the way to the end without worrying about others.

Reply:
Done, thanks!